From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Levy <tomlevy93@gmail.com>
Date: Sun, 5 Sep 2021 02:36:59 +1200
Subject: backport: Modify domjudge:db-config:update command

Clarify documentation, display a clear startup message, fix typos.

Reduce the logging verbosity from 'info' to 'debug' for messages about
configuration options that have non-default values. (It can be useful
to see which options have been modified, but it adds too much noise
because Docker containers run this command on startup.)
---
 .../src/Command/UpdateDatabaseConfigurationCommand.php | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/webapp/src/Command/UpdateDatabaseConfigurationCommand.php b/webapp/src/Command/UpdateDatabaseConfigurationCommand.php
index b9aa7cf01..16d8cf31a 100644
--- a/webapp/src/Command/UpdateDatabaseConfigurationCommand.php
+++ b/webapp/src/Command/UpdateDatabaseConfigurationCommand.php
@@ -69,8 +69,8 @@ class UpdateDatabaseConfigurationCommand extends Command
     {
         $this
             ->setName('domjudge:db-config:update')
-            ->setDescription('Update the database configuration from etc/db-config.yaml')
-            ->setHelp('Helps backport configuration changes from DOMjudge 7.3 to 7.2.')
+            ->setDescription('Update the configuration options in the database to match etc/db-config.yaml')
+            ->setHelp('Helps backport configuration changes from DOMjudge 7.3 to 7.2 by creating new options in the database to match options added to db-config.yaml (which was introduced in 7.3).')
             ->addOption('dry-run', null, InputOption::VALUE_NONE, 'Show what would be done without modifying database');
     }
 
@@ -85,12 +85,14 @@ class UpdateDatabaseConfigurationCommand extends Command
             $output->setVerbosity(OutputInterface::VERBOSITY_VERY_VERBOSE);
         }
 
-        // Load db-config.yaml (see ConfigurationService.php on DOMjudge 7.3)
+        // Load db-config.yaml (see ConfigurationService.php in DOMjudge 7.3)
         $yamlDbConfigFile = $this->etcDir . '/db-config.yaml';
         $fileLocator      = new FileLocator($this->etcDir);
         $loader           = new YamlConfigLoader($fileLocator);
         $yamlConfig       = $loader->load($yamlDbConfigFile);
 
+	$this->logger->notice('Updating the configuration options in the database to match etc/db-config.yaml');
+
         $count = 0;
         foreach ($yamlConfig as $category) {
             foreach ($category['items'] as $item) {
@@ -127,7 +129,7 @@ class UpdateDatabaseConfigurationCommand extends Command
                                 $this->em->flush();
                             }
                         } else {
-                            $this->logger->info("Configuration option '%s': value in database is %s, default value db-config.yaml is %s", [$item['name'], OutputFormatter::escape($this->dj->jsonEncode($currentValue)), OutputFormatter::escape($this->dj->jsonEncode($item['default_value']))]);
+                            $this->logger->debug("Configuration option '%s': value in database is %s, default value in db-config.yaml is %s", [$item['name'], OutputFormatter::escape($this->dj->jsonEncode($currentValue)), OutputFormatter::escape($this->dj->jsonEncode($item['default_value']))]);
                         }
                     }
                     if ($option->getPublic() !== $item['public']) {
-- 
2.20.1

